custom/icds/translations/integrations/transifex.py

```diff
@@ -83,3 +88,68 @@ def _cleanup(self):
         for resource_name, filepath in self.po_file_generator.generated_files:
             if os.path.exists(filepath):
                 os.remove(filepath)
+
+    def _get_resource_slugs_for_version(self):
+        """
+        :return: list of resource slugs corresponding to version
+        """
+        return [r['name']
+                for r in self.client.list_resources().json()
+                if r['name'].endswith("v%s" % self.version)]
+
+    def _ensure_resources_belong_to_version(self, resource_slugs):
+        """
+        confirms that resource slugs provided are for the expected version by checking for its name to end with
+        v[version number] like v15 for version 15.
+        :param resource_slugs: list of resource slugs
+        """
+        for resource_slug in resource_slugs:
+            if not resource_slug.endswith("v%s" % self.version):
+                raise Exception("Resource name '{}' is expected to contain version".format(
+                    resource_slug
+                ))
+
+    def get_translations(self, resource_slugs=None):
+        """
+        pull translations from transifex
+        :param resource_slugs: optional argument. All resource slugs corresponding to version are used otherwise.
+        :return: dict of resource_slug mapped to POEntry objects
+        """
+        # if passed ensure belonging to version
+        if resource_slugs:
+            self._ensure_resources_belong_to_version(resource_slugs)
+        else:
+            resource_slugs = self._get_resource_slugs_for_version()
+        # if still no resources raise error
+        if not resource_slugs:
+            raise Exception("No resources found for this version")
+        po_entries = {}
+        for resource_slug in resource_slugs:
+            po_entries[resource_slug] = self.client.get_translation(resource_slug, self.source_lang,
+                                                                    self.lock_translations)
+        return po_entries
+
+    def resources_pending_translations(self, break_if_true=False):
```

Comment:
This method returns a single slug if `break_if_true`, and a list of slugs if `break_if_false`.  It looks like this is used in only one place, and it's being used as a boolean.  It feels to me like it's one method trying to act like several depending on how it's called.  Considering this is only used in a boolean context, and with `break_if_true=True`, I'd change the name and implementation to reflect that.  That's my preference, but I'll defer to you on the final call.

Reply:
> it's being used as a boolean

the result is used to list the pending resources/resources as well, so `resources_pending_translations` sounds fine. Can make it always return a list, so in case `break_if_true` is `True` return a list with the single resource to make the return predictable.